lib/paths.ml

```diff
@@ -66,9 +66,9 @@ struct
     | a :: q -> a :: chop_extension q
 
   let parse_filename name =
-    let l = Support.split_on_char (String.get (Filename.dir_sep) 0) name in
-    match List.rev l with
-    | "" :: q -> List.rev q
+    Support.cuts ~empty:true ~sep:Filename.dir_sep name
+    |> List.rev |> function
+    | "" :: l -> List.rev l
```

Comment:
On this patch, I kept the same behavior as before. The cleaned way will be to trim only one side of the given list - and we probably can implement such function into the `Support` module. Tell me if you need such function or not. We probably should explicit the behavior of this function too or manipulate an abstract type such as `Fpath.t`. But it's another question which should be asked into another issue 🙂